app/models/trip.rb

```diff
@@ -0,0 +1,20 @@
+class Trip < ApplicationRecord
+  has_many :tickets, dependent: :nullify
+  belongs_to :bus, optional: true
+  has_many :seats, dependent: :nullify
+  belongs_to :route, optional: true
+  has_many :boardings, dependent: :nullify
+  has_many :boarding_points, through: :boardings
```

Comment:
what is through representing?

Reply:
By using `through`, we are implementing `has_many_through` association.
Here, `Trip` model has many `Boarding Points` through `Boarding` Model.